plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java

```diff
@@ -0,0 +1,129 @@
+import org.opensearch.ml.helper.ModelAccessControlHelper;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.search.fetch.subphase.FetchSourceContext;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class GetModelGroupTransportAction extends HandledTransportAction<ActionRequest, MLModelGroupGetResponse> {
+
+    Client client;
+    NamedXContentRegistry xContentRegistry;
+    ClusterService clusterService;
+
+    ModelAccessControlHelper modelAccessControlHelper;
+
+    @Inject
+    public GetModelGroupTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry,
+        ClusterService clusterService,
+        ModelAccessControlHelper modelAccessControlHelper
+    ) {
+        super(MLModelGroupGetAction.NAME, transportService, actionFilters, MLModelGroupGetRequest::new);
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+        this.clusterService = clusterService;
+        this.modelAccessControlHelper = modelAccessControlHelper;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLModelGroupGetResponse> actionListener) {
+        MLModelGroupGetRequest mlModelGroupGetRequest = MLModelGroupGetRequest.fromActionRequest(request);
+        String modelGroupId = mlModelGroupGetRequest.getModelGroupId();
+        FetchSourceContext fetchSourceContext = getFetchSourceContext(mlModelGroupGetRequest.isReturnContent());
+        GetRequest getRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId).fetchSourceContext(fetchSourceContext);
+        User user = RestActionUtils.getUserContext(client);
+
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            ActionListener<MLModelGroupGetResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
+            client.get(getRequest, ActionListener.wrap(r -> {
+                if (r != null && r.isExists()) {
+                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+
+                        MLModelGroup mlModelGroup = MLModelGroup.parse(parser);
+                        modelAccessControlHelper.validateModelGroupAccess(user, modelGroupId, client, ActionListener.wrap(access -> {
+                            if (!access) {
+                                wrappedListener
+                                    .onFailure(
+                                        new MLValidationException(
```

Comment:
Let's add this [type of exception](https://github.com/opensearch-project/ml-commons/blob/feature/agent_framework_dev/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java#L112-L116) as this shouldn't be a 500 level error.

Reply:
Sure will do